File: Modules/HumanResource/OrganisationHierarchy/Section/SectionForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.OrganisationHierarchy.Forms;

[FormScript("OrganisationHierarchy.Section")]
[BasedOnRow(typeof(SectionRow), CheckNames = true)]
public class SectionForm
{
    public string Name { get; set; }
    public string Description { get; set; }
}

File: Modules/ServerTypes/OrganisationHierarchy/SectionForm.ts
import { StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface SectionForm {
    Name: StringEditor;
    Description: StringEditor;
}

export class SectionForm extends PrefixedContext {
    static formKey = 'OrganisationHierarchy.Section';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SectionForm.init)  {
            SectionForm.init = true;

            var w0 = StringEditor;

            initFormType(SectionForm, [
                'Name', w0,
                'Description', w0
            ]);
        }
    }
}